dmci/distributors/pycsw_dist.py

```diff
@@ -0,0 +1,188 @@
+        status = self._get_transaction_status(self.TOTAL_INSERTED, resp)
+
+        return status
+
+    def _update(self):
+        """ Update current entry
+
+        Need to find out how to do this...
+        """
+        logger.warning("Not yet implemented")
+
+        return False
+
+    def _delete(self):
+        if self._metadata_id is None:
+            msg = "Metadata identifier must be a non-empty string"
+            logger.error(msg)
+            return False
+
+        headers = requests.structures.CaseInsensitiveDict()
+        headers["Content-Type"] = "application/xml"
+        headers["Accept"] = "application/xml"
+        xml_as_string = ('<?xml version="1.0" encoding="UTF-8"?>'
+                         '<csw:Transaction xmlns:ogc="http://www.opengis.net/ogc" '
+                         'xmlns:csw="http://www.opengis.net/cat/csw/2.0.2" '
+                         'xmlns:ows="http://www.opengis.net/ows" '
+                         'xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" '
+                         'xsi:schemaLocation="http://www.opengis.net/cat/csw/2.0.2 '
+                         'http://schemas.opengis.net/csw/2.0.2/CSW-publication.xsd" '
+                         'service="CSW" version="2.0.2">'
+                         '   <csw:Delete>'
+                         '       <csw:Constraint version="1.1.0">'
+                         '           <ogc:Filter>'
+                         '               <ogc:PropertyIsEqualTo>'
+                         '                   <ogc:PropertyName>apiso:Identifier</ogc:PropertyName>'
+                         '                   <ogc:Literal>%s</ogc:Literal>'
+                         '               </ogc:PropertyIsEqualTo>'
+                         '           </ogc:Filter>'
+                         '       </csw:Constraint>'
+                         '   </csw:Delete>'
+                         '</csw:Transaction>' % self._metadata_id)
+        resp = requests.post(self._conf.csw_service_url, headers=headers, data=xml_as_string)
+        status = self._get_transaction_status(self.TOTAL_DELETED, resp)
+
+        return status
+
+    def _get_transaction_status(self, key, resp):
+        """ Check response status, read response text, and get status (boolean)
+
+        Input
+        =====
+        key : str
+            total_inserted, total_updated or total_deleted
+        resp : requests.models.Response
+            Response object from transaction
+
+        Return
+        ======
+        status : boolean
+            True if the transaction succeeded, otherwise False
+        """
+        if key not in self.STATUS:
+            logger.error("Input key must be '%s', '%s' or '%s'" % (self.TOTAL_INSERTED,
+                                                                   self.TOTAL_UPDATED,
+                                                                   self.TOTAL_DELETED))
+            return False
+        if not type(resp) is requests.models.Response:
```

Comment:
This is fine as long as you only accept exact type match. If the object can be inherited, you must instead use:
```python
if not isinstance(resp, requests.models.Response):
```

But regardless, the line should at least be rewritten to:
```python
if type(resp) is not requests.models.Response:
```
As @johtoblan  pointed out.

Reply:
Solved in 29ab426b602c87447e96f6ffa09585a049c715d5